File: app/cqrs/mr/readModel/detailsview/InventoryItemDeactivatedHandler.java
package cqrs.mr.readModel.detailsview;

import cqrs.core.Handler;
import cqrs.mr.events.InventoryItemDeactivated;
import cqrs.mr.readModel.BullShitDatabase;

public class InventoryItemDeactivatedHandler implements Handler<InventoryItemDeactivated> {

	@Override
	public void handle(InventoryItemDeactivated message) {
		BullShitDatabase.details().remove(message.id());
	}
}

File: app/cqrs/mr/readModel/detailsview/InventoryItemDeactivatedHandler.scala
package cqrs.mr.readModel.detailsview

import cqrs.core.Handler
import cqrs.mr.events.InventoryItemDeactivated
import cqrs.mr.readModel.BullShitDatabase

class InventoryItemDeactivatedHandler extends Handler[InventoryItemDeactivated] {

	def handle(message: InventoryItemDeactivated) {
		BullShitDatabase.details.remove(message.id)
	}
}